rust_src/src/buffers.rs

```diff
@@ -871,6 +871,57 @@ pub fn force_mode_line_update(all: LispObject) -> LispObject {
     all
 }
 
+/// Return a list of overlays which is a copy of the overlay list
+/// LIST, but for buffer BUFFER.
+#[no_mangle]
+pub extern "C" fn copy_overlays(
+    buffer: *mut Lisp_Buffer,
+    list: *mut Lisp_Overlay,
+) -> *mut Lisp_Overlay {
+    if list == ptr::null_mut() {
+        return list;
```

Comment:
I think we don't need this NULL check since the function returns NULL if there are no overlays ? Btw we use `is_null` in this situation.

Reply:
I realized you have to distinguish between NULL pointers and invalid overlay references.